app/main.py

```diff
@@ -1,8 +1,30 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        self.people[name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    def __repr__(self) -> str:
+        return f"<Person name={self.name}>"
+
+
+def create_person_list(people_data: list) -> list:
+    person_list = []
+    for person_dict in people_data:
```

Comment:
You can do this in two loops. And simplify code.
1. First loop for create all person
2. Second loop for set wife/husband